src/components/fx/hover.js

```diff
@@ -772,7 +772,10 @@ function _hover(gd, evt, subplot, noHoverEvent) {
         hoverdistance: fullLayout.hoverdistance
     };
 
-    var hoverLabels = createHoverText(hoverData, labelOpts, gd);
+    var actualHoverData = hoverData.filter(function(d) {
```

Comment:
Shouldn't this line be ```hoverData = hoverData.filter(function(d) { ```?
Also wondering if this kind of filtering should be performed earlier in the process e.g. right after (or at) the call to 
```
findHoverPoints();
```

Reply:
No that does not work because the hover event must still be propagated to the in hover callback. The problem is still that one test darks because it expects an empty hover for the test but it bails out early because nothing is shown for that. Could you take a look at this?

Reply:
@archmoj

Reply:
If I'm reading the test failure correctly, seems like `createHoverText(actualHoverData, ...)` fails when there's only `hoverinfo: 'none'` in the remaining points. @Domino987 I think you're right to create a separate `actualHoverData` but we need to guard the `createHoverText` call against this case. Something like:

```js
var hoverLabels = actualHoverData.length && createHoverText(...)

// keep us from dropping into the next block that's all about arranging these labels
if (hoverLabels && !helpers.isUnifiedHover(hovermode)) {
```

Reply:
Yes that sounds like a good idea. I implemented it but unfortunately it did not fix the tests. Not sure what is causing the errors now though. Weird.

Reply:
Could you please fetch `upstream/master` in your fork then, pull the master and merge it into your branch?
That should fix some weird ones.